fusion-engine-driver/utils/conversion_utils.hpp

```diff
@@ -220,4 +236,38 @@ class ConversionUtils {
     nmea.sentence = nmea_stream.str();
     return nmea;
   }
+
+  static const std::unordered_map<int, std::string>& getLeapSeconds() {
+    static const std::unordered_map<int, std::string> leap_seconds_ = {
+      {10, "01 Jan 1972"},
+      {11, "01 Jul 1972"},
+      {12, "01 Jan 1973"},
+      {13, "01 Jan 1974"},
+      {14, "01 Jan 1975"},
+      {15, "01 Jan 1976"},
```

Comment:
Only store a number, like: 
```   
{23, 1985},
{24, 1988},
{25, 1990},
```

Reply:
You do not want to use years as the key. Leap seconds can occur at any time, but by convention happen on midnight of January 1 or July 1, or both. So you can't assume it takes effect January 1 and just key by year, and technically there can even be two leap seconds in one year.

Typically, leap second tables are keyed by a full date/time in seconds corresponding with midnight when the new leap second takes effect. Either a UTC time, or a TAI or GPS time. Depending on which you use, you need to be careful to handle the transition correctly. See https://www.ietf.org/timezones/data/leap-seconds.list, which is the official leap second

One additional feature: you might also want to think about downloading the leap second table automatically, or using the one that's already available on the computer. See  https://git.ligo.org/cds/software/gpstime/-/blob/master/gpstime/leaps.py for reference. We use that library in our Python code. It loads the leap second table from the OS, or downloads it if it doesn't exist yet.

Reply:
Is it necessary to make a similar system by retrieving the file via the Internet if it doesn't exist on my bone, or to retrieve it directly via my computer? Or is it conceivable to keep the map system but with timestamps?

Reply:
I think a fixed map is ok here. This won't change frequently, and we can push an update when it does. Leap seconds are announced 6 months in advance.